docs/reference/hooks.rst

```diff
@@ -1379,8 +1379,27 @@ Audit log
         def additional_log_actions(actions):
             actions.register_action('wagtail_package.echo', _('Echo'), _('Sent an echo'))
 
-            def callback_message(data):
-                return _('Hello %(audience)s') % {
-                    'audience': data['audience'],
-                }
-            actions.register_action('wagtail_package.with_callback', _('Callback'), callback_message)
+
+    Alternatively, for a log message that varies according to the log entry's data, create a subclass of ``wagtail.core.log_actions.LogFormatter`` that overrides the ``format_message`` method, and use ``register_action`` as a decorator on that class:
+
+    .. code-block:: python
+
+        from django.utils.translation import gettext_lazy as _
+
+        from wagtail.core import hooks
+        from wagtail.core.log_actions import LogFormatter
+
+        @hooks.register('register_log_actions')
+        def additional_log_actions(actions):
+            @actions.register_action('wagtail_package.greet_audience')
```

Comment:
Any reason to not allow this at the module level? This would be consistent with how [modeladmin classes are registered](https://docs.wagtail.io/en/stable/reference/contrib/modeladmin/index.html#a-more-complicated-example)

Reply:
No reason that I can see. This is the "eliminate hooks that just register something on startup" task that I put forward for the forthcoming Python refactoring sprint, and `register_log_actions` is on my hit list for that :-)

(you may have noticed that I pointedly avoided creating a new hook for admin URL finders in #7504, in favour of just calling the register function at the module level - so if that doesn't blow up, it should serve as a testing-the-water exercise in removing them more widely!)